src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs

```diff
@@ -18,13 +22,26 @@ private class ParsedSyntaxTree : CSharpSyntaxTree
             private readonly bool _hasCompilationUnitRoot;
             private readonly Encoding _encodingOpt;
             private readonly SourceHashAlgorithm _checksumAlgorithm;
+            private readonly ImmutableDictionary<string, ReportDiagnostic> _diagnosticOptions;
             private SourceText _lazyText;
 
-            internal ParsedSyntaxTree(SourceText textOpt, Encoding encodingOpt, SourceHashAlgorithm checksumAlgorithm, string path, CSharpParseOptions options, CSharpSyntaxNode root, Syntax.InternalSyntax.DirectiveStack directives, bool cloneRoot = true)
+            internal ParsedSyntaxTree(
+                SourceText textOpt,
+                Encoding encodingOpt,
+                SourceHashAlgorithm checksumAlgorithm,
+                string path,
+                CSharpParseOptions options,
+                CSharpSyntaxNode root,
+                Syntax.InternalSyntax.DirectiveStack directives,
+                bool cloneRoot = true,
+                ImmutableDictionary<string, ReportDiagnostic> diagnosticOptions = null)
             {
                 Debug.Assert(root != null);
                 Debug.Assert(options != null);
                 Debug.Assert(textOpt == null || textOpt.Encoding == encodingOpt && textOpt.ChecksumAlgorithm == checksumAlgorithm);
+                Debug.Assert(diagnosticOptions is null ||
+                    diagnosticOptions.IsEmpty ||
+                    diagnosticOptions.KeyComparer == StringOrdinalComparer.Instance);
```

Comment:
what's the reasoning here?

Reply:
It protects us from forgetting to pass our own comparer.